### ISSUE TITLE
Add possibility to chose different service engines for some of the services.

### ISSUE THREAD

**Issue by mdolling-gfz** [research-software-directory/research-software-directory#765]
Because different organizations might want to use different services for
- login
- database
- backup

it would be helpful to make the services "variable".
The most simple solution would be, to generate the compose file with python or bash magic (`generate_service_stack.sh`, `make`) or even other solutions.

**Comment by jspaaks** [comment]
Yes different options are generally good, but I can see downsides as well, for example be aware of the consequences 
- for the readability of the documentation: text with a lot of conditionals get hard to read for users, as well as maintainers
- for maintenance: more degrees of freedom means exponential more things to check

With that in mind,
1. offering different login mechanism doesnt affect the rest of the code too much, so would be OK
2. I think I wouldn't offer the opportunity to switch out Mongo for say mysql (but perhaps you could run a Mongo database some other place than what's there now)
3. backup service is already delegating to a command which you could tweak. BTW the command is Xenon, an abstraction library for a bunch of storage and compute backends. Xenon already allows you to switch to say an SFTP server, as well as some other types, see https://github.com/xenon-middleware/xenon

**Comment by mdolling-gfz** [comment]
the first concrete action would be the login. turned out we do just oauth2. i will start implementing a general oauth2 client, implement it into the existing login and make an ENV var to switch service (with github as default)
